File: src/168_excel_sheet_column_title/excel_sheet_column_title.cpp
#include <algorithm>
#include <iostream>
#include <string>
using namespace std;

class Solution {
   public:
    string convertToTitle(int columnNumber) {
        string res = "";
        while (columnNumber > 0) {
            int offset = (columnNumber - 1) % 26;
            res += char('A' + offset);
            columnNumber = (columnNumber - 1) / 26;
        }
        reverse(res.begin(), res.end());
        return res;
    }
};

int main() {
    Solution s;
    cout << s.convertToTitle(1) << " | A" << endl;
    cout << s.convertToTitle(28) << " | AB" << endl;
    cout << s.convertToTitle(701) << " | ZY" << endl;
    cout << s.convertToTitle(52) << " | AZ" << endl;
    cout << s.convertToTitle(2147483647) << " | FXSHRXW" << endl;
    return 0;
}

File: src/168_excel_sheet_column_title/ExcelSheetColumnTitle.java
public class ExcelSheetColumnTitle {
    public String convertToTitle(int columnNumber) {
        StringBuilder res = new StringBuilder();
        while (columnNumber > 0) {
            int offset = (columnNumber - 1) % 26;
            res.append((char) ('A' + offset));
            columnNumber = (columnNumber - 1) / 26;
        }
        return res.reverse().toString();
    }

    public static void main(String[] args) {
        ExcelSheetColumnTitle s = new ExcelSheetColumnTitle();
        System.out.println(s.convertToTitle(1) + " | A");
        System.out.println(s.convertToTitle(28) + " | AB");
        System.out.println(s.convertToTitle(701) + " | ZY");
        System.out.println(s.convertToTitle(52) + " | AZ");
        System.out.println(s.convertToTitle(2147483647) + " | FXSHRXW");
    }
}
